### ISSUE TITLE
Error installing distribution

### ISSUE THREAD

**Issue by Dlazder** [Magisk-Modules-Alt-Repo/chroot-distro#6]
When installing the distribution I received these errors:
```
.../files/home # chroot-distro install kali
grep: warning: stray \ before /
mount: /data/local/chroot-distro/kali/dev: mount point does not exist.
       dmesg(1) may have more information after failed mount system call.
mount: /data/local/chroot-distro/kali/sys: mount point does not exist.
       dmesg(1) may have more information after failed mount system call.
mount: /data/local/chroot-distro/kali/proc: mount point does not exist.
       dmesg(1) may have more information after failed mount system call.
mount: /data/local/chroot-distro/kali/dev/pts: mount point does not exist.
       dmesg(1) may have more information after failed mount system call.
/system/bin/chroot-distro[844]: can't create /data/local/chroot-distro/kali/etc/resolv.conf: No such file or directory
/system/bin/chroot-distro[844]: can't create /data/local/chroot-distro/kali/etc/hosts: No such file or directory
chroot: failed to run command ‘/bin/su’: No such file or directory
.../files/home #
```

I mounted the system partition with write permissions, but this did not change nothing.

**Comment by jjkola** [comment]
I have fixed install and uninstall related bugs in the PR which is mentioned before. Afterwards I did fix backup and restore functionality but I will not create a PR for it until #7 has been merged as the new fix is built on top of the PR. I would strongly recommend using code from my repository (fix_backup_and_restore branch) until all fixes has been merged. Otherwise people will have to be prepared for manual fix-ups (and possible device wipes, if using version without my fixes, well, they are avoidable but great care has to be taken).

**Comment by Dlazder** [comment]
@jjkola 

Same error

```
.../files/home # chroot-distro reinstall kali
grep: warning: stray \ before /
mount: /data/local/chroot-distro/kali/dev: mount point does not exist.
       dmesg(1) may have more information after failed mount system call.
mount: /data/local/chroot-distro/kali/sys: mount point does not exist.
       dmesg(1) may have more information after failed mount system call.
mount: /data/local/chroot-distro/kali/proc: mount point does not exist.
       dmesg(1) may have more information after failed mount system call.
mount: /data/local/chroot-distro/kali/dev/pts: mount point does not exist.
       dmesg(1) may have more information after failed mount system call.
/system/bin/chroot-distro[844]: can't create /data/local/chroot-distro/kali/etc/resolv.conf: No such file or directory
/system/bin/chroot-distro[844]: can't create /data/local/chroot-distro/kali/etc/hosts: No such file or directory
chroot: failed to run command ‘/bin/su’: No such file or directory
```

**Comment by Dlazder** [comment]
Now I tried another distribution
```
~ $ sudo chroot-distro install ubuntu
grep: warning: stray \ before /
/bin/su: error while loading shared libraries: libdl.so: cannot open shared object file: No such file or directory
```

**Comment by jjkola** [comment]
Hmm, sounds like we will have to investigate further. What is your device / environment?

**Comment by Dlazder** [comment]
> Hmm, sounds like we will have to investigate further. What is your device / environment?

My device poco m5, hyperOS 1.0.9.0 ULUMIXM, android 14

**Comment by jjkola** [comment]
Ok, what about Magisk and busybox-ndk? And what is your command line? termux? or something different?

**Comment by Dlazder** [comment]
> Ok, what about Magisk and busybox-ndk? And what is your command line? termux? or something different?

Yes, I use Termux. busybox-ndk installed.
![Screenshot_2024-07-12-18-49-19-629_com.topjohnwu.magisk.jpg](https://github.com/user-attachments/assets/b9f4c561-538d-4b4c-9d4e-94dda0686675)

**Comment by jjkola** [comment]
That is too old version of busybox-ndk. You should update it to latest one (1.36.1) as even 1.32.1 has some troubles (at least with wget).

**Comment by Dlazder** [comment]
> That is too old version of busybox-ndk. You should update it to latest one (1.36.1) as even 1.32.1 has some troubles (at least with wget).

Maybe I'm not paying attention, but I couldn't find the new magisk module. Can you help me with this?

**Comment by jjkola** [comment]
busybox-ndk does provide updateJson property in module.prop file so if the packge manager you use supports it then it should update to the latest one automatically. Unfortunately newer versions of Magisk no longer provide download based updates so either you have to do so manually or rely on separate package managers. updateJson points currently to https://raw.githubusercontent.com/Magisk-Modules-Repo/busybox-ndk/master/update.json from where you can find zipUrl property with the latest install package.

**Comment by Dlazder** [comment]
> busybox-ndk does provide updateJson property in module.prop file so if the packge manager you use supports it then it should update to the latest one automatically. Unfortunately newer versions of Magisk no longer provide download based updates so either you have to do so manually or rely on separate package managers. updateJson points currently to https://raw.githubusercontent.com/Magisk-Modules-Repo/busybox-ndk/master/update.json from where you can find zipUrl property with the latest install package.

![IMG_20240712_220249.jpg](https://github.com/user-attachments/assets/3dfa6acb-2fc0-4dce-88df-175b18bfe7fa)

Unfortunately, updating the busybox version did not help.

**Comment by Dlazder** [comment]
I was able to log into 2 distributions, although I couldn’t work with them. 
![IMG_20240714_200411.jpg](https://github.com/user-attachments/assets/24604d1a-48af-4c69-a054-62b6f505e934)

![IMG_20240714_200323.jpg](https://github.com/user-attachments/assets/dc2e06ef-e1ca-477e-80ec-ecdc09e2aac3)

**Comment by jjkola** [comment]
You have clearly installed them properly since you have now root@localhost prompts in both of them.

In the first screenshot you have:
- error: could not determine cachedir mount point /var/cache/packman/pkg <-- This seems like you will have to consult arch linux documentation how to resolve this
- error: failed to commit transaction (not enough free disk space) <-- This sounds like you have too little space available on the device? Is there some dangling installations or something else eating the space? Maybe you should move the `/data/local/chroot-distro/` folder to external SD card and then mount it at `/data/local/chroot-distro`?

The second screenshot:
Before installing anything I would suggest updating the package information with `apt update` and then upgrade old components with `apt upgrade`, and only after this I would install whatever tools I need. And if that does not work then I would use the tools available with basic installation (maybe vi or something else) to investigate why thing(s) does not work.

**Comment by jjkola** [comment]
If you still have 'chroot: failed to run command ‘/bin/su’: No such file or directory' error with some distro then please provide with which exact rootfs do you have a problem and before reporting problems, please test with latest main branch from this repo. If you still have same problem after installing latest development version (and rebooting afterwards), I will investigate the problem. The problems with rootfs have to be investigated with the respective maintainers unless it is clearly something caused by chroot-distro script. Please report back so that we can close the issue if the original issue is no longer problem.

**Comment by Dlazder** [comment]
> You have clearly installed them properly since you have now root@localhost prompts in both of them.
> 
> In the first screenshot you have:
> - error: could not determine cachedir mount point /var/cache/packman/pkg <-- This seems like you will have to consult arch linux documentation how to resolve this
> - error: failed to commit transaction (not enough free disk space) <-- This sounds like you have too little space available on the device? Is there some dangling installations or something else eating the space? Maybe you should move the `/data/local/chroot-distro/` folder to external SD card and then mount it at `/data/local/chroot-distro`?
> 
> The second screenshot:
> Before installing anything I would suggest updating the package information with `apt update` and then upgrade old components with `apt upgrade`, and only after this I would install whatever tools I need. And if that does not work then I would use the tools available with basic installation (maybe vi or something else) to investigate why thing(s) does not work.

The problem was that when updating Ubuntu, the connection to the server did not occur. Now for some reason it works, I successfully installed several packages. In the case of Arch, I fixed this error and got a new one, I think this true arch experience) These problems are most likely not related to chroot-distro. I haven't tested all distributions, but Kali still gives the same error

```
.../files/home # chroot-distro login kali
mount: /data/local/chroot-distro/kali/dev: mount point does not exist.
       dmesg(1) may have more information after failed mount system call.
mount: /data/local/chroot-distro/kali/sys: mount point does not exist.
       dmesg(1) may have more information after failed mount system call.
mount: /data/local/chroot-distro/kali/proc: mount point does not exist.
       dmesg(1) may have more information after failed mount system call.
mount: /data/local/chroot-distro/kali/dev/pts: mount point does not exist.
       dmesg(1) may have more information after failed mount system call.
chroot: failed to run command ‘/bin/su’: No such file or directory
```

**Comment by Dlazder** [comment]
How can I download the latest version of the module? There seems to be an old version in the releases.

**Comment by jjkola** [comment]
Please check report #9 for instructions on how to create a package zip. Please note, that you will have to reboot the device before the new version can be used.

I'll try to add instructions to documentation later but for now this will have to suffice. Also, I think it is time to create a new release shortly as there hasn't been any findings with the new code and it fixes several problems with original code.

If the issue still persist with latest code then please provide the exact rootfs you have downloaded so that I can try replicating the issue.

**Comment by Dlazder** [comment]
> Please check report #9 for instructions on how to create a package zip. Please note, that you will have to reboot the device before the new version can be used.
> 
> I'll try to add instructions to documentation later but for now this will have to suffice. Also, I think it is time to create a new release shortly as there hasn't been any findings with the new code and it fixes several problems with original code.
> 
> If the issue still persist with latest code then please provide information about the exact rootfs you have downloaded so that I can try replicating the issue.

I installed the new module. At this point I was able to install and login in kali without errors.  When I try to log into CentOS I get the error /bin/su. I also want to point out an unusual problem. Packages are downloaded only when using mobile Internet. When connected to Wi-Fi, I cannot connect to the repositories. This problem exists on Kali and Ubuntu. Perhaps this does not apply to the chroot-distro, but suddenly you know how to fix it?

**Comment by jjkola** [comment]
Thanks for the information. I will check it when I have time. I have a hunch that decompressing was not successful but because there is currently no checks for that it will silently fail and manifest itself with some random/semi-random/specific error depending on how much it was able to decompress.

**Comment by jjkola** [comment]
Checked the problem and it was not what I expected, created a new one: #20

**Comment by jjkola** [comment]
Closing this as resolved. If there is still issues with installing (other than with CentOS) please create a new issue.
